src/ui/main.cpp

```diff
@@ -166,6 +166,8 @@ int main(int argc, char* argv[]) {
         return exportSound(maybeArgs.value());
     }
 
+    cli.~QCoreApplication();
```

Comment:
Hu, that's scary :) You can avoid this by storing the QCoreApplication in a std::unique_ptr. You would need to `#include <memory>`, then start of `main()` would go like this:

```cpp
auto cli = std::make_unique<QCoreApplication>(argc, argv);
QCommandLineParser parser;
setupCommandLineParser(&parser);
parser.process(cli.get);
// ...
```

Then to delete our instance:

```cpp
cli.reset();
```

Reply:
I have to admit I don't see what's scary about calling a destructor or how this wrapper makes things better.

Reply:
What scares me is the idea of having an instance hanging around after its destructor has been called. Admittedly it's not used here, but I don't like it.

Reply:
This stackoverflow answer gives more reasons why it's not a good idea: https://stackoverflow.com/q/65797371

Reply:
Okay, thanks for the explanation.